eth/tracers/api_test.go

```diff
@@ -323,6 +334,18 @@ func TestTraceCall(t *testing.T) {
 			expectErr: fmt.Errorf("block #%d %w", genBlocks+1, ethereum.NotFound),
 			//expect:    nil,
 		},
+		// Optimism: Trace block on the historical chain
+		{
+			blockNumber: rpc.BlockNumber(777),
+			call: ethapi.TransactionArgs{
+				From:  &accounts[0].addr,
+				To:    &accounts[1].addr,
+				Value: (*hexutil.Big)(big.NewInt(1000)),
+			},
+			config:    nil,
+			expectErr: nil,
+			expect:    `{"gas":21000,"failed":false,"returnValue":"777","structLogs":[]}`,
+		},
```

Comment:
This fails on line 411 below: 

https://github.com/ethereum-optimism/op-geth/blob/82841479add528a945271528236c087bb0954232/eth/tracers/api_test.go#L410-L412

    api_test.go:411: test 3: result mismatch, want {"gas":21000,"failed":false,"returnValue":"777","structLogs":[]}, got {"gas":0,"failed":false,"returnValue":"","structLogs":null}
    
The really strange thing is that the `have` value looks right, but has nil values: 

`{"gas":0,"failed":false,"returnValue":"","structLogs":null}`

My suspicion is that something wrong in the handling and marshalling of the `histResult` in here: 
https://github.com/ethereum-optimism/op-geth/blob/82841479add528a945271528236c087bb0954232/eth/tracers/api.go#L876-L893